cscs-checks/apps/greasy/greasy_check.py

```diff
@@ -0,0 +1,218 @@
+                                                  msg=f'Rank id {rank(x)} is '
+                                                  'not lower than the number'
+                                                  'of ranks '
+                                                  f'{self.nranks_per_worker} in'
+                                                  ' output file '), result),
+                    sn.map(lambda x: sn.assert_lt(tid(x),
+                                                  self.num_cpus_per_task,
+                                                  msg=f'Rank id {tid(x)} is not'
+                                                  ' lower than the number of '
+                                                  'cpus per task '
+                                                  f'{self.num_cpus_per_task} in'
+                                                  f' output file {output_file}'),
+                           result),
+                    sn.map(lambda x: sn.assert_eq(num_threads(x),
+                                                  num_cpus_per_task,
+                                                  msg=f'Found {num_threads(x)} '
+                                                      'threads rather '
+                                                      'than '
+                                                      f'{self.num_cpus_per_task}'
+                                                      ' in output file '
+                                                      f'{output_file}'), result),
+                    sn.map(lambda x: sn.assert_lt(rank(x), num_tasks,
+                                                  msg=f'Found {rank(x)} threads'
+                                                  ' rather than '
+                                                  f'{self.num_cpus_per_task} in'
+                                                  f' output file {output_file}'),
+                           result),
+                    sn.map(lambda x: sn.assert_eq(num_ranks(x), num_tasks,
+                                                  msg=f'Number of ranks '
+                                                      f'{num_ranks(x)} is not '
+                                                      'equal to '
+                                                      f'{self.nranks_per_worker}'
+                                                      ' in output file '
+                                                      f'{output_file}'), result),
+                )
+            ))
+
+        sn.evaluate(sn.assert_found(r'Finished greasing', self.greasy_logfile))
+
+        sn.evaluate(sn.assert_found(f'INFO: Summary of {self.num_greasy_tasks} '
+                                    f'tasks: '
+                                    f'{self.num_greasy_tasks} OK, '
+                                    f'0 FAILED, '
+                                    f'0 CANCELLED, '
+                                    f'0 INVALID\.',
```

Comment:
To avoid PEP8 issues:

```suggestion
                                    fr'0 INVALID\.',
```